src/Mapping/Annotations/AbstractDocument.php

```diff
@@ -4,6 +4,14 @@
 
 namespace Doctrine\ODM\MongoDB\Mapping\Annotations;
 
-abstract class AbstractDocument implements Annotation
-{
+use function class_exists;
+
+class_exists(\Doctrine\ODM\MongoDB\Mapping\Attribute\AbstractDocument::class);
+
+// @phpstan-ignore-next-line if.alwaysFalse
+if (false) {
+    /** @deprecated Use \Doctrine\ODM\MongoDB\Mapping\Attribute\AbstractDocument instead */
+    abstract class AbstractDocument extends \Doctrine\ODM\MongoDB\Mapping\Attribute\AbstractDocument
```

Comment:
The new class no longer implements the `Annotation` class. We can fix this for the static analysis class, but it won't work for the `class_alias` declaration in the attribute.

```suggestion
    abstract class AbstractDocument extends \Doctrine\ODM\MongoDB\Mapping\Attribute\AbstractDocument implements Annotation
```

Reply:
The abstract class cannot be used as an annotation. Did you find an issue?

Reply:
This might create a BC break for people with a custom document annotation, but I'd argue that it's very unlikely someone would do that. If you're fine with the potential BC break, so am I :)

Reply:
I added a note to the backlog and marked the new abstract attribute classes as internal.